include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,648 @@
+
+    /// @copydoc value_type
+    using const_reference = char;
+
+    /** The unsigned integer type used to represent size.
+    */
+    using size_type = std::size_t;
+
+    /** The signed integer type used to represent differences.
+    */
+    using difference_type = std::ptrdiff_t;
+
+    /** A read-only bidirectional iterator to the decoded range.
+
+        This is a read-only bidirectional iterator to
+        the decoded characters.
+
+    */
+#ifdef BOOST_URL_DOCS
+    using iterator = __see_below__;
+#else
+    class iterator;
+#endif
+
+    /// @copydoc const_iterator
+    using const_iterator = iterator;
+
+    /** A reverse read-only bidirectional iterator to the decoded range.
+
+        This is a read-only bidirectional iterator to
+        the decoded characters.
+
+    */
+    using reverse_iterator = std::reverse_iterator<iterator>;
+
+    /// @copydoc reverse_iterator
+    using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+
+    /** Constructor
+
+        Default-constructed objects represent
+        the empty string.
+
+    */
+    decoded_view(
+        pct_decode_opts const& opt = {}) noexcept
+        : p_(nullptr)
+        , n_(0)
+        , dn_(0)
+        , plus_to_space_(opt.plus_to_space)
+    {}
+
+    /** Constructor
+    */
+    BOOST_URL_DECL
+    decoded_view(
+        string_view str,
+        pct_decode_opts opt = {}) noexcept;
+
+    /** Returns an iterator to the beginning
+    */
+    const_iterator
+    begin() const noexcept;
+
+    /** Returns an iterator to the end
+    */
+    const_iterator
+    end() const noexcept;
+
+    /** Accesses the first decoded character
+
+        @note The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+    */
+    const_reference
+    front() const;
+
+    /** Accesses the last decoded character
+
+        @note The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+    */
+    const_reference
+    back() const;
+
+    /** Returns the encoded string view
+
+        This function returns a string_view of
+        the underlying encoded string.
+
+        @return The encoded string view.
+    */
+    string_view
+    encoded() const
+    {
+        return {p_, n_};
+    }
+
+    /** Returns the number of char elements in the string
+    */
+    size_type
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Returns whether the string is empty
+    */
+    bool
+    empty() const noexcept
+    {
+        return n_ == 0;
+    }
+
+    /** Copies a decoded substring to another character string
+
+        Copies a substring to the character array pointed to by
+        the destination char string, where `rcount` is the smaller
+        of `count` and `size() - pos`.
+
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid positions.
+
+        @throw std::out_of_range If `pos > size()`
```

Comment:
How does this render?

Reply:
<div class="informaltable"><table class="table">
<colgroup>
<col>
<col>
</colgroup>
<thead><tr>
<th>
                  <p>
                    Type
                  </p>
                </th>
<th>
                  <p>
                    Thrown On
                  </p>
                </th>
</tr></thead>
<tbody><tr>
<td>
                  <p>
                    <code class="computeroutput"><span class="identifier">std</span><span class="special">::</span><span class="identifier">out_of_range</span></code>
                  </p>
                </td>
<td>
                  <p>
                    If <code class="computeroutput"><span class="identifier">pos</span> <span class="special">&gt;</span>
                    <span class="identifier">size</span><span class="special">()</span></code>
                  </p>
                </td>
</tr></tbody>
</table></div>

Reply:
That's not a screen capture you just made it up